src/common.h

```diff
@@ -369,6 +369,36 @@ char *ly_strnchr(const char *s, int c, size_t len);
  */
 int ly_strncmp(const char *refstr, const char *str, size_t str_len);
 
+/**
+ * @brief Get all possible value prefixes from an YANG value by iteratively returning specific substrings.
+ *
+ * The function looks for possible prefix ending in a colon at the beginning of @p str_begin.
+ * If @p str_begin does not contain the prefix at the beginning, then either:
+ * 1. Returns the entire input string if the input string does not contain the prefix at all.
+ * 2. Returns a substring before the prefix.
+ *
+ * Examples of inputs and outputs are shown in the table below.
+ * Output string @p str_next is used in the next iteration as input parameter @p str_begin.
+   @verbatim
+   |            INPUT              |                         OUTPUT                        |
+   |                               |   iteration 1    |   iteration 2    |   iteration 3   |
+   |------------------------------ |------------------|------------------|-----------------|
+   | /namespace_prefix:some_string | /                | namespace_prefix | some_string     |
+   | namespace_prefix:some_string  | namespace_prefix | some_string      | NULL            |
+   | /some_string                  | /some_string     | NULL             | NULL            |
+   @endverbatim
```

Comment:
I miss the example regarding the case 2 described above, probably this (not sure if it is correct, especially regarding the semicolon character):
```
| string1;string2     | string1;  | string2 | NULL
```

Reply:
Semicolon is not treated specially in any way, there is nothing in the description hinting this, right? It should simply be
```
| string1;string2     | string1;string2 | NULL
```
Right @lePici ?

Reply:
Yes only the colon is taken into account. I will add information about the colon to the documentation. But if there is a use-case with a semicolon somewhere, I can extend the function specification.

Reply:
No, to my knowledge it would be redundant to do anything with a semicolon so I do not know why @rkrejci asked about it.

Reply:
ok, what about `string1;prefix:something`? That's the case that matters, right? My point is that point 2 in the description talks about some substring before the prefix, but it is not specified how the substring and prefix are delimited.

Reply:
and it's not only semicolon, it is recognized using `is_xmlqnamechar()`

Reply:
Oh, okay, this information can be added, I suppose. Can you suggest something concise? The description was a bit general because I do not think you need to know the exact delimiting characters to be able to use it, you just know that you have a YANG value and want to get the prefixes from it, so you will use this function. Also, for this to work perfectly the input format should probably matter as in JSON the prefixes can only be YANG module names meaning a subset of the characters allowed for XML. But even if something else is returned for JSON, it will never match a module name, so no harm done.

Reply:
ok, I'll add this comment to the description.